### DISCUSSION TITLE
Modify WrapPanel wrap policy?

### DISCUSSION THREAD

**Discussion by TheRealQuantam** [AvaloniaUI/Avalonia#11410] [Q&A] [answered]
I'm looking at WrapPanel as a possible solution to a problem. However, I see that when wrapping vertically (which is what I need) its policy appears to be to wrap rows that do not fit in the previous column at all. Even if only the top 1 or 2 pixels of the row is visible it will not wrap. Is there a way to modify this so that it will wrap any row that is not fully (or some other fraction) visible in the previous column?

**Comment by timunie** [comment]
@TheRealQuantam I'm suprised about what you write above. Can you do me 2 favors?
1. tell us the Avalonia version you use
2. provide the code used and images showing the issue. Best if a minimal sample can be shared

**Comment by TheRealQuantam** [comment]
The version is 0.10.19. Here is an image of what I ran into while looking to convert a StackPanel to WrapPanel. There is an option hiding at the bottom of the left side, only a couple pixels visible which causes it not to be wrapped; if the window is shrunk a couple more pixels it will wrap to the right side. 

Note that I didn't write the original GUI code and I'm not particularly familiar with Avalonia, I'm just trying to solve the problem that the window got too tall; I just Googled a bit to find out about WrapPanel.

![image](https://github.com/AvaloniaUI/Avalonia/assets/54358053/64df9a2a-a0e2-4ddd-84fc-81c95df2801d)

**Reply by timunie** [reply]
I'm unable to reproduce with 0.10.19 and 11.0.0p8. I think there may be somewhere a height or a Margin set, but without a minimal sample I cannot help you further. Check the XAML and maybe use Devtools [F12] to check the layout. 

![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/56565962-29d6-474a-8e32-7f00813b9635)

**Reply by TheRealQuantam** [reply, marked as answer]
HMMM. That's a lot more information than I understand, but it looks like if I increase the bottom margin on the WrapPanel to 16 (it was 0 on all sides) it solves the problem. That will do. Thanks.
